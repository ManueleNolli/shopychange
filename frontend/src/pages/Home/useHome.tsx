import { useEffect, useState } from 'react'
import {
  handlerQueryErrorCatch,
  salesQuery,
} from '../../services/BackendService'
import { useUserContext } from '../../context/userContext'
import { NFT } from '../../types/components/NFT'
import useFetching from '../../hooks/useFetching'

export default function useHome() {
  const { blockchainNetworkId, userAddress } = useUserContext()
  const [sales, setSales] = useState<NFT[]>([])
  const { isFetching, isError, setIsFetching, setIsError } = useFetching()

  useEffect(() => {
    const fetchSales = async () => {
      if (!blockchainNetworkId) return

      const response = await handlerQueryErrorCatch(
        salesQuery(blockchainNetworkId)
      )

      if (response.error) {
        setIsError(true)
        setIsFetching(false)
        return
      }

      const sales = response.result.sales as NFT[]
      setSales(sales)
      setIsFetching(false)
    }

    fetchSales()
  }, [blockchainNetworkId])

  return { sales, isFetching, isError, isConnected: userAddress !== null }
}
