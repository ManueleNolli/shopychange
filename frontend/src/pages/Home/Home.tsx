import React from 'react'
import { Box, Center, Heading, Text } from '@chakra-ui/react'
import NFTOrCollectionViewer from '../../components/NFTOrCollectionViewer/NFTOrCollectionViewer'
import QueryContainer from '../../utils/QueryContainer/QueryContainer'
import useHome from './useHome'

export default function Home() {
  const { sales, isFetching, isError, isConnected } = useHome()

  return (
    <Box>
      <Center>
        <Heading>Market</Heading>
      </Center>
      {isConnected ? (
        <QueryContainer isLoading={isFetching} isError={isError}>
          <Center>
            <NFTOrCollectionViewer nfts={sales} />
          </Center>
        </QueryContainer>
      ) : (
        <Center>
          <Text mt="10">Please connect your wallet to view the market</Text>
        </Center>
      )}
    </Box>
  )
}
