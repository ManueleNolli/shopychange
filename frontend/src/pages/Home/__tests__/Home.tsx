import React from 'react'
import { render } from '@testing-library/react'
import Home from '../Home'
import useHome from '../useHome'

jest.mock('../../../components/NFTOrCollectionViewer/NFTOrCollectionViewer')

jest.mock('../useHome', () => jest.fn())

describe('Home', () => {
  test('renders successfully when not connected', () => {
    ;(useHome as jest.Mock).mockReturnValue({
      sales: [],
      isFetching: false,
      isError: false,
      isConnected: false,
    })
    const tree = render(<Home />)
    expect(tree.container).toMatchSnapshot()
    expect(
      tree.getByText('Please connect your wallet to view the market')
    ).toBeInTheDocument()
  })
  test('renders successfully when connected', () => {
    ;(useHome as jest.Mock).mockReturnValue({
      sales: [],
      isFetching: false,
      isError: false,
      isConnected: true,
    })
    const tree = render(<Home />)
    expect(tree.container).toMatchSnapshot()
    expect(tree.getByTestId('NFTOrCollectionViewer')).toBeInTheDocument()
  })
})
