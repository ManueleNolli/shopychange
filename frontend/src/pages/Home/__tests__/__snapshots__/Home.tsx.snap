// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home renders successfully when connected 1`] = `
<div>
  <div
    class="css-0"
  >
    <div
      class="css-gmuwbf"
    >
      <h2
        class="chakra-heading css-0"
      >
        Market
      </h2>
    </div>
    <div
      class="css-gmuwbf"
    >
      <div
        data-testid="NFTOrCollectionViewer"
      />
    </div>
  </div>
</div>
`;

exports[`Home renders successfully when not connected 1`] = `
<div>
  <div
    class="css-0"
  >
    <div
      class="css-gmuwbf"
    >
      <h2
        class="chakra-heading css-0"
      >
        Market
      </h2>
    </div>
    <div
      class="css-gmuwbf"
    >
      <p
        class="chakra-text css-1q7njkh"
      >
        Please connect your wallet to view the market
      </p>
    </div>
  </div>
</div>
`;
